readme.md

```diff
@@ -7,67 +7,257 @@
-  * [Polaris API Key](#polaris-api-key)
-  * [Dependencies](#dependencies)
-  * [Building with CMake](#building-with-cmake)
-  * [Building With Bazel](#building-with-bazel)
-    * [Cross-Compilig With Bazel](#cross-compiling-with-bazel)
-  * [Building On Mac](#building-on-mac)
-* [Example Applications](#example-applications)
-  * [NTRIP Proxy Example](#ntrip-proxy-example)
-  * [Septentrio Example](#septentrio-example)
-    * [Hardware Setup](#hardware-setup)
-    * [Configure AsteRx-m2](#configure-asterx-m2)
-    * [Serial Permissions in Ubuntu](#serial-permissions-in-ubuntu)
-    * [Running the example](#running-the-example)
-    * [Verifying Corrections](#verifying-corrections)
-  * [Generic Serial Receiver Example](#generic-serial-receiver-example)
-
-## Usage ##
-
-The Polaris Client can be built by Bazel or CMake. Follow instructions below for build system of your choice.
+### Table Of Contents ###
+
+* [Polaris C Client](#polaris-c-client)
+  * [Requirements](#requirements)
+  * [Building From Source](#building-from-source)
+    * [Bazel](#bazel)
+      * [Compiling Example Applications With Bazel](#compiling-example-applications-with-bazel)
+    * [CMake](#cmake)
+    * [GNU Make](#gnu-make)
+  * [Using Polaris C Client](#using-polaris-c-client)
+* [Polaris C++ Client](#polaris-c-client-1)
+  * [Requirements](#requirements-1)
+  * [Building From Source](#building-from-source-1)
+    * [Bazel](#bazel-1)
+      * [Compiling Example Applications With Bazel](#compiling-example-applications-with-bazel-1)
+      * [Cross-Compiling With Bazel](#cross-compiling-with-bazel)
+    * [CMake](#cmake-1)
+    * [Building On Mac OS](#building-on-mac-os)
+  * [Using Polaris C++ Client](#using-polaris-c-client-1)
+  * [Example Applications](#example-applications)
+    * [Simple Polaris Client](#simple-polaris-client)
+    * [Generic Serial Receiver Example](#generic-serial-receiver-example)
+    * [NTRIP Proxy Example](#ntrip-proxy-example)
+    * [Septentrio Example](#septentrio-example)
+      * [Hardware Setup](#hardware-setup)
+      * [Configure AsteRx-m2](#configure-asterx-m2)
+      * [Serial Permissions in Ubuntu](#serial-permissions-in-ubuntu)
+      * [Running the example](#running-the-example)
+      * [Verifying Corrections](#verifying-corrections)
 
 ### Polaris API Key ###
 
-To establish a connection, you must provide a valid Polaris API key. Please contact the administrator of your Point One Navigation contract or sales@pointonenav.com if you do not have one.
+To establish a connection, you must provide a valid Polaris API key. Please contact the administrator of your Point One
+Navigation contract or sales@pointonenav.com if you do not have one.
 
+## Polaris C Client ##
 
-### Building with CMake ###
+### Requirements ###
 
-To build using CMake, install the dependencies if you do not have them already.
+- [Bazel](https://bazel.build/) 3.3+, or [CMake](https://cmake.org/) 3.3+ and
+  [GNU Make](https://www.gnu.org/software/make/)
```

Comment:
technically we support Visual studio as well, right?

Reply:
No? We don't have a Visual Studio project defined anywhere.